### PR TITLE
Upgrade ESLint to version 4

### PR DESCRIPTION
**Pull request by ChristianMurphy**
Linter will now error if it receives an invalid configuration [[1]](http://eslint.org/docs/user-guide/migrating-to-4.0.0#config-validation), ESLint now is better able to handle comments [[2]](http://eslint.org/docs/user-guide/migrating-to-4.0.0#-ast-nodes-no-longer-have-comment-properties), Spacing rules are now stricter by default [[3]](http://eslint.org/docs/user-guide/migrating-to-4.0.0#indent-rewrite) [[4]](http://eslint.org/docs/user-guide/migrating-to-4.0.0#padded-blocks-defaults) [[5]](http://eslint.org/docs/user-guide/migrating-to-4.0.0#-the-space-before-function-paren-rule-is-more-strict-by-default) [[6]](http://eslint.org/docs/user-guide/migrating-to-4.0.0#-the-no-multi-spaces-rule-is-more-strict-by-default).

----

Contributor License Agreement adherence:

- [x] This Contribution is under the terms of Individual [Contributor License Agreements][] (and also Corporate Contributor License Agreements to the extent applicable) appearing in the [Apereo CLA roster][].

[Apereo CLA roster]: http://licensing.apereo.org/completed-clas
[Contributor License Agreements]: https://www.apereo.org/licensing/agreements

### CODE REVIEW

File: angularjs-portal-home/src/main/webapp/my-app/marketplace/controllers.js [hunk 1]
```diff
@@ -235,10 +235,11 @@ define(['angular', 'jquery', 'require'], function(angular, $, require) {
...
           currentPage = 'market';
-          if(currentCategory !== '')
-            $scope.categoryToShow = currentCategory;
-          else
-            $scope.categoryToShow = '';
+          if (currentCategory !== '') {
+$scope.categoryToShow = currentCategory;
+} else {
+$scope.categoryToShow = '';
+}
```
Comment (davidmsibley) [inline angularjs-portal-home/src/main/webapp/my-app/marketplace/controllers.js]: what happened to this spacing?

**Comment by davidmsibley** [comment]
I'm starting to see the value of those [Greenkeeper-like products](https://github.com/UW-Madison-DoIT/angularjs-portal/issues/639)

File: angularjs-portal-home/src/main/webapp/my-app/marketplace/controllers.js (hunk 1 above)
Comment (ChristianMurphy) [inline angularjs-portal-home/src/main/webapp/my-app/marketplace/controllers.js]: ESLint fixer went in to fix this area, [eslint-config-google currently has the indent rule disabled](https://github.com/google/eslint-config-google/blob/699a43f7408160980f445fdce236110b367c2159/index.js#L199) :disappointed: 
So the fixer did not appropriately resolve.

Comment (ChristianMurphy) [inline angularjs-portal-home/src/main/webapp/my-app/marketplace/controllers.js]: I added `intent` set for 2 spaces to warn level and ran autofix.

**Comment by davidmsibley** [comment]
eh, I'm not big on that indent rule if it doesn't necessarily get us closer to google standard.  I'd rather just manually fix the 4 lines from the previous commit.

**Comment by ChristianMurphy** [comment]
@davidmsibley It does get it closer to the Google code style, with the exception it does not yet have the ability to check continuation lines (which are ignored rather than errored).

**Comment by davidmsibley** [comment]
Thanks, I was daunted by the 2000 line changes for that little change.

**Comment by ChristianMurphy** [comment]
@apetro @davidmsibley @vertein are there any additional changes needed for this PR to be merged?

**Comment by davidmsibley** [comment]
nope, just needed to be nagged

**Pull request merged**